apps/src/blockly/googleBlocklyWrapper.ts

```diff
@@ -755,16 +755,18 @@ function initializeBlocklyWrapper(blocklyInstance: GoogleBlocklyInstance) {
 
     blocklyWrapper.setMainWorkspace(workspace);
 
-    // Same flyout callbacks are used for both main workspace categories
-    // and categories when modal function editor is enabled.
-    workspace.registerToolboxCategoryCallback(
-      'PROCEDURE',
-      functionsFlyoutCategory
-    );
-    workspace.registerToolboxCategoryCallback(
-      'VARIABLE',
-      variablesFlyoutCategory
-    );
+    if (!optOptionsExtended.useBlocklyDynamicCategories) {
+      // Same flyout callbacks are used for both main workspace categories
+      // and categories when modal function editor is enabled.
+      workspace.registerToolboxCategoryCallback(
+        'PROCEDURE',
+        functionsFlyoutCategory
+      );
+      workspace.registerToolboxCategoryCallback(
+        'VARIABLE',
+        variablesFlyoutCategory
+      );
+    }
```

Comment:
@mikeharv This appeared to be necessary since our implementations don't yet work for non-string toolbox definitions, per:
https://github.com/code-dot-org/code-dot-org/blob/7728bbaa9a8c21bf46ef3fd1a10fcc0cc4382b7d/apps/src/blockly/addons/cdoUtils.ts#L398-L403

Reply:
That's fair.  That said, could it be more future-proof to use Blockly's implementation rather than our own "fork" of it, if we don't need to use anything in our fork?

Reply:
The problem is that Blockly's implementation doesn't do what we need it to. If we just used Blockly's implementation, old levels would be missing blocks in the variables category. 
Blockly's implementation (defined [here](https://github.com/google/blockly/blob/84218fe9b1c82e4795cdd4c8e74aa73aac18a1aa/core/variables.ts#L84-L106)) will always populate the category with the button and a dynamic set of blocks (`variables_set`, `math_change`, and `variables_get`) based on the variables currently defined on the workspace. Our levels require that we do that and additionally include blocks from the xml for whatever lesson-specific context applies (e.g. prompt blocks or standard blocks with pre-set initial values).

Reply:
Oh sorry, I meant in the case of Music Lab, where we don't need these extra features thus far.

Reply:
Oh, you were probably talking about my suggestion re: Functions, then? I guess I don't have a strong enough opinion then. My instinct is still that internal consistency would be better. I don't worry so much about our custom logic breaking Music Lab because we already depend on it for all other labs. If we think it's easy for other engineers to understand that Music Lab is an exception here that's fine with me.

Reply:
Ah yes, I should have been more clear.  Thanks!